src/pytest_workflow/file_tests.py

```diff
@@ -105,20 +118,23 @@ def repr_failure(self, excinfo):
 
 class FileMd5(pytest.Item):
     def __init__(self, parent: pytest.Collector, filepath: Path,
-                 md5sum: str):
+                 md5sum: str, workflow: Workflow):
         """
         Create a tests for the file md5sum.
         :param parent: The collector that started this item
         :param filepath: The path to the file
         :param md5sum:  The expected md5sum
+        :param workflow: The workflow running to generate the file
         """
         name = "md5sum"
         super().__init__(name, parent)
         self.filepath = filepath
         self.expected_md5sum = md5sum
         self.observed_md5sum = None
+        self.workflow = workflow
 
     def runtest(self):
+        self.workflow.wait()
```

Comment:
Why `wait` here while `ContetentTestItem` uses `join`? Same question for the `FileExists` class.

Reply:
`wait` is a function of `Workflow`. `join()` is a function of thread.